### ISSUE TITLE
Logs to Google Cloud Storage: workers don't write but UI tries to read from Storage

### ISSUE THREAD

**Issue by informatica92** [bitnami/bitnami-docker-airflow#43]
**Description**

Trying to set up aurflow in order to write logs to Google Cloud Storage instead of writing them on the local drive, I added the following lines to the docked-compose-yml file in the airflow, airflow-worker and airflow-scheduler section:
```
      - AIRFLOW__CORE__REMOTE_LOGGING=True
      - AIRFLOW__CORE__REMOTE_BASE_LOG_FOLDER=gs://airflow_[...]/logs/
      - AIRFLOW__CORE__REMOTE_LOG_CONN_ID=google_cloud
```
**Steps to reproduce the issue:**

1. set up docker-compose-yml file as shown above
2. restarted services 
3. run a dag

**Describe the results you received:**

I got this
```
*** Unable to read remote log from gs://airflow_[...]/logs/hello_world/sample-task/2020-05-14T14:23:11.363283+00:00/1.log
*** 'NoneType' object has no attribute 'download'

*** Log file does not exist: /opt/bitnami/airflow/logs/hello_world/sample-task/2020-05-14T14:23:11.363283+00:00/1.log
*** Fetching from: http://7134dc48d12c:8793/log/hello_world/sample-task/2020-05-14T14:23:11.363283+00:00/1.log

[2020-05-14 14:23:24,982] {taskinstance.py:669} INFO - Dependencies all met for <TaskInstance: hello_world.sample-task 2020-05-14T14:23:11.363283+00:00 [queued]>
[2020-05-14 14:23:25,016] {taskinstance.py:669} INFO - Dependencies all met for <TaskInstance: hello_world.sample-task 2020-05-14T14:23:11.363283+00:00 [queued]>
[2020-05-14 14:23:25,016] {taskinstance.py:879} INFO - 
--------------------------------------------------------------------------------
[2020-05-14 14:23:25,016] {taskinstance.py:880} INFO - Starting attempt 1 of 1
[2020-05-14 14:23:25,017] {taskinstance.py:881} INFO - 
--------------------------------------------------------------------------------
[2020-05-14 14:23:25,027] {taskinstance.py:900} INFO - Executing <Task(HelloOperator): sample-task> on 2020-05-14T14:23:11.363283+00:00
[2020-05-14 14:23:25,031] {standard_task_runner.py:53} INFO - Started process 88 to run task
[2020-05-14 14:23:25,136] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: hello_world.sample-task 2020-05-14T14:23:11.363283+00:00 [running]> 7134dc48d12c
[2020-05-14 14:23:25,166] {logging_mixin.py:112} INFO - Hello foo_bar
[2020-05-14 14:23:25,200] {taskinstance.py:1065} INFO - Marking task as SUCCESS.dag_id=hello_world, task_id=sample-task, execution_date=20200514T142311, start_date=20200514T142324, end_date=20200514T142325
[2020-05-14 14:23:34,928] {logging_mixin.py:112} INFO - [2020-05-14 14:23:34,928] {local_task_job.py:103} INFO - Task exited with return code 0
```
Moreover, the bucked doesn't contain any files.

**Describe the results you expected:**

I expected to see log files into the bucket and to read them from airflow UI but this is not what's happening.

**Additional information you deem important (e.g. issue happens only occasionally):**

Also it seems that logs are not written in the correct location but the UI tries to read logs from there (so it looks that the UI is configured properly but workers dont't write logs in the correct position).
It is also strange that logs are normally written locally as you can see above (it seems that workers completely ignore the setting) 

**Version**

- Output of `docker version`:

```
Client:
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.13.8
 Git commit:        afacb8b7f0
 Built:             Wed Mar 11 23:42:35 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       afacb8b7f0
  Built:            Wed Mar 11 22:48:33 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu2
  GitCommit:
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:
 docker-init:
  Version:          0.18.0
  GitCommit:

```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 5
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: btrna5ha16okj6003f4h88295
  Is Manager: true
  ClusterID: mj08kpmf7o5xyyr0e94cjgy84
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 10.132.0.20
  Manager Addresses:
   10.132.0.20:2377
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version:
 runc version:
 init version:
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-1009-gcp
 Operating System: Ubuntu 20.04 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 3.597GiB
 Name: airflow-docker
 ID: CVAT:NDID:XDYQ:M7MA:DN2Z:WW2J:5MDW:HHGO:AGB2:45WH:H7HF:NMZD
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support

```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.8.2
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020

```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Moreover, the service account I am using correctly reads and writes from/to the bucket (I tried to perform a test using command line inside the containers)

**Comment by carrodher** [comment]
Hi, thanks for reporting this issue, I am going to try to reproduce it. In the meantime, can you check if the env. variables are properly propagated to the container? For that, you can execute
```
docker exec -it CONTIANER_ID env | grep REMOTE
```
where `CONTIANER_ID` is the identifier of the container obtained with `docker ps`.

According to the [official documentation](https://airflow.readthedocs.io/en/stable/howto/set-config.html#setting-configuration-options):
> The universal order of precedence for all configuration options is as follows:
> - set as an environment variable
> - set as a command environment variable
> - set in airflow.cfg
> - command in airflow.cfg
> - Airflow’s built in defaults

so it seems if the env. variables are properly set the configuration should be taken into account. In the same way, I just found this documentation about [how to configure writing logs to GCS](https://airflow.readthedocs.io/en/stable/howto/write-logs.html#writing-logs-to-google-cloud-storage), did you follow those steps? Please note that `pip install 'apache-airflow[gcp]'` is not necessary as it is already installed in the image.

**Comment by informatica92** [comment]
Hi @carrodher, thanks for the quick reply.
```
user@airflow-docker:~/bitnami-airflow$ sudo docker exec -it bitnami-airflow_airflow-scheduler_1 env | grep REMOTE
AIRFLOW__CORE__REMOTE_LOGGING=True
AIRFLOW__CORE__REMOTE_BASE_LOG_FOLDER=gs://airflow_[...]/logs/
AIRFLOW__CORE__REMOTE_LOG_CONN_ID=google_cloud
user@airflow-docker:~/bitnami-airflow$ sudo docker exec -it bitnami-airflow_airflow-worker_1 env | grep REMOTE
AIRFLOW__CORE__REMOTE_LOGGING=True
AIRFLOW__CORE__REMOTE_BASE_LOG_FOLDER=gs://airflow_[...]/logs/
AIRFLOW__CORE__REMOTE_LOG_CONN_ID=google_cloud
user@airflow-docker:~/bitnami-airflow$ sudo docker exec -it bitnami-airflow_airflow_1 env | grep REMOTE
AIRFLOW__CORE__REMOTE_LOGGING=True
AIRFLOW__CORE__REMOTE_BASE_LOG_FOLDER=gs://airflow_[...]/logs/
AIRFLOW__CORE__REMOTE_LOG_CONN_ID=google_cloud
```
As you can see from the snippet above, the env variables are correctly propagated (they are only missing in 'postgres' and 'redis') 

About the apache-airflow[gcp] installation, I added a requirements.txt file and I mounted it in /bitnami/python/ adding this line to the yml file: 
```
  airflow-scheduler:
    [...]
    volumes:
      [...]
      - ./python:/bitnami/python
      [...]

  airflow-worker:
    [...]
    volumes:
      [...]
      - ./python:/bitnami/python
      [...]

  airflow:
    [...]
    volumes:
      [...]
      - ./python:/bitnami/python
      [...]

```
and inside this folder, the requirements.txt file contains:
`apache-airflow[gcp]`

Anyway this only actually works with `airflow` becasue the its `run.sh` is the only one that takes the requirements.txt file into consideration.
But I also tried to manually install `apache-airflow[gcp]` library in the `scheduler `and `worker `container but nothing changes

**Comment by carrodher** [comment]
Not sure if this is going to fix the issue, but we just released a new version of the image (`1.10.10-debian-10-r30`) as we detected that we were installing `apache-airflow[gcp-api]` that seems to be an old method, being the new one `apache-airflow[gcp]`. Can you try using this new image? You can use the specific tag `1.10.10-debian-10-r30` or one of the rolling tags associated with this one like `latest`, `1`, `1.10.10`, `1-debian-10`, etc.

If you are using the docker-compose present in the root of this repository just ensure a new image is downloaded, this is the difference between the previous one and the new one:
```
▶ docker images
REPOSITORY          TAG                     IMAGE ID            CREATED             SIZE
bitnami/airflow     1.10.10-debian-10-r30   8cf99f55ab7e        5 hours ago         1GB
bitnami/airflow     1.10.10-debian-10-r29   96fd5f24a896        35 hours ago        968MB
```
now it is 30MB bigger because we added/updated all the extra-packages listed [here](https://airflow.apache.org/docs/stable/installation.html#extra-packages).

**Comment by informatica92** [comment]
Hi @carrodher,
this morning I updated the images. (I deleted everything with a system prune before just to be sure) 
```
m.riezzo@airflow-docker:~/airflow-docker$ sudo docker ps
CONTAINER ID        IMAGE                         COMMAND                  CREATED             STATUS              PORTS                    NAMES
9a072bcba51c        bitnami/redis:5.0             "/opt/bitnami/script…"   27 minutes ago      Up 27 minutes       6379/tcp                 airflow-docker_redis_1
63c2d5494d55        bitnami/postgresql:10         "/opt/bitnami/script…"   27 minutes ago      Up 27 minutes       5432/tcp                 airflow-docker_postgresql_1
d98cbc3dc038        bitnami/airflow-scheduler:1   "/app-entrypoint.sh …"   27 minutes ago      Up 27 minutes                                airflow-docker_airflow-scheduler_1
b76d826349d2        bitnami/airflow-worker:1      "/app-entrypoint.sh …"   27 minutes ago      Up 27 minutes       8793/tcp                 airflow-docker_airflow-worker_1
22e610b9c8a8        bitnami/airflow:1             "/app-entrypoint.sh …"   27 minutes ago      Up 27 minutes       0.0.0.0:8080->8080/tcp   airflow-docker_airflow_1
```

And now it works! 🥳 

![image](https://user-images.githubusercontent.com/17550961/82194041-7fb27500-98f6-11ea-8204-e6e651c084a7.png)


I think that I can close the issue now.
THANK YOU!
